### PR TITLE
UPDATE Jenkinsfile for parallel stage execution

### PR DESCRIPTION
**Pull request by robnester-rh**
Updated the Jenkinsfile and associated library files to allow testing
stages to run in parallel.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -322,107 +322,119 @@ podTemplate(name: podName,
...
+                                pipelineUtils.executeInContainer(currentStage, "ostree-boot-image", "/home/ostree-boot-image.sh")
+
+                                // Set our message topic, properties, and content
+                                messageFields = pipelineUtils.setMessageFields("package.test.functional.queued")
+
+                                // Send message org.centos.prod.ci.pipeline.package.test.functional.queued on fedmsg
+                                pipelineUtils.sendMessageWithAudit(messageFields['properties'], messageFields['content'], msgAuditFile, fedmsgRetryCount)
+                            }
+                        }
+                    }, functionalj-tests {
```
Comment (johnbieren) [inline Jenkinsfile]: typo

File: Jenkinsfile [hunk 2]
```diff
@@ -322,107 +322,119 @@ podTemplate(name: podName,
...
+                                // Run ostree boot sanity
+                                env.rsync_paths = "images"
+                                env.rsync_from = "${env.RSYNC_USER}@${env.RSYNC_SERVER}::${env.RSYNC_DIR}/${env.branch}/"
+                                env.rsync_to = "/home/output/"
+                                pipelineUtils.executeInContainer(currentStage + "-rsync-before", "rsync", "/tmp/rsync.sh")
+
+                                pipelineUtils.executeInContainer(currentStage, "ostree-boot-image", "/home/ostree-boot-image.sh")
+
+                                // Set our message topic, properties, and content
+                                messageFields = pipelineUtils.setMessageFields("package.test.functional.queued")
```
Comment (johnbieren) [inline Jenkinsfile]: Gotta move all the .queued messages to right before parallel step

**Review by johnbieren — CHANGES_REQUESTED** [review]

**Review by johnbieren — APPROVED** [review]
lgtm

**Comment by scoheb** [comment]
Lgtm

**Comment by robnester-rh** [comment]
[test]

**Comment by robnester-rh** [comment]
[test]

**Comment by robnester-rh** [comment]
Just a note, this latest push is meant for testing purposes; no need to review code at the moment.

File: Jenkinsfile [hunk 3]
```diff
@@ -90,62 +90,62 @@ podTemplate(name: podName,
         containers: [
                 // This adds the custom slave container to the pod. Must be first with name 'jnlp'
                 containerTemplate(name: 'jnlp',
-                        image: DOCKER_REPO_URL + '/' + OPENSHIFT_NAMESPACE + '/jenkins-continuous-infra-slave:' + SLAVE_TAG,
```
Comment (scoheb) [inline Jenkinsfile]: you do not like my distinct touch here? ;)

**Review by scoheb — APPROVED** [review]

**Comment by robnester-rh** [comment]
[test]

**Comment by robnester-rh** [comment]
[test]

**Comment by robnester-rh** [comment]
[test]

**Comment by robnester-rh** [comment]
[test]

**Comment by robnester-rh** [comment]
[test]

**Comment by robnester-rh** [comment]
[test]

**Comment by robnester-rh** [comment]
[test]

**Comment by robnester-rh** [comment]
[test]

**Pull request closed without merging**